lxd/daemon.go

```diff
@@ -632,88 +658,89 @@ func startDaemon() (*Daemon, error) {
-			c := &lxd.Config{Remotes: map[string]lxd.RemoteConfig{}}
-			_, err := lxd.NewClient(c, "")
-			if err != nil {
-				shared.Log.Debug("Detected stale unix socket, deleting")
-				// Connecting failed, so let's delete the socket and
-				// listen on it ourselves.
-				err = os.Remove(localSocketPath)
+			// If the socket exists, let's try to connect to it and see if there's
+			// a lxd running.
+			if shared.PathExists(localSocketPath) {
+				c := &lxd.Config{Remotes: map[string]lxd.RemoteConfig{}}
+				_, err := lxd.NewClient(c, "")
 				if err != nil {
-					return nil, err
+					shared.Log.Debug("Detected stale unix socket, deleting")
+					// Connecting failed, so let's delete the socket and
+					// listen on it ourselves.
+					err = os.Remove(localSocketPath)
+					if err != nil {
+						return err
+					}
 				}
 			}
-		}
-
-		unixAddr, err := net.ResolveUnixAddr("unix", localSocketPath)
-		if err != nil {
-			return nil, fmt.Errorf("cannot resolve unix socket address: %v", err)
-		}
 
-		unixl, err := net.ListenUnix("unix", unixAddr)
-		if err != nil {
-			return nil, fmt.Errorf("cannot listen on unix socket: %v", err)
-		}
-
-		if err := os.Chmod(localSocketPath, 0660); err != nil {
-			return nil, err
-		}
+			unixAddr, err := net.ResolveUnixAddr("unix", localSocketPath)
+			if err != nil {
+				return fmt.Errorf("cannot resolve unix socket address: %v", err)
+			}
 
-		gid, err := shared.GroupId(*group)
-		if err != nil {
-			return nil, err
-		}
+			unixl, err := net.ListenUnix("unix", unixAddr)
+			if err != nil {
+				return fmt.Errorf("cannot listen on unix socket: %v", err)
+			}
 
-		if err := os.Chown(localSocketPath, os.Getuid(), gid); err != nil {
-			return nil, err
-		}
+			if err := os.Chmod(localSocketPath, 0660); err != nil {
+				return err
+			}
 
-		sockets = append(sockets, unixl)
-	}
+			gid, err := shared.GroupId(*group)
+			if err != nil {
+				return err
+			}
 
-	listenAddr, err := d.ConfigValueGet("core.https_address")
-	if err != nil {
-		return nil, err
-	}
+			if err := os.Chown(localSocketPath, os.Getuid(), gid); err != nil {
+				return err
+			}
 
-	if listenAddr != "" {
-		_, _, err := net.SplitHostPort(listenAddr)
-		if err != nil {
-			listenAddr = fmt.Sprintf("%s:%s", listenAddr, shared.DefaultPort)
+			sockets = append(sockets, unixl)
 		}
 
-		tcpl, err := tls.Listen("tcp", listenAddr, tlsConfig)
+		listenAddr, err := d.ConfigValueGet("core.https_address")
 		if err != nil {
-			return nil, fmt.Errorf("cannot listen on https socket: %v", err)
+			return err
 		}
 
-		sockets = append(sockets, tcpl)
+		if listenAddr != "" {
+			_, _, err := net.SplitHostPort(listenAddr)
+			if err != nil {
+				listenAddr = fmt.Sprintf("%s:%s", listenAddr, shared.DefaultPort)
+			}
+
+			tcpl, err := tls.Listen("tcp", listenAddr, tlsConfig)
+			if err != nil {
+				return fmt.Errorf("cannot listen on https socket: %v", err)
+			}
+
+			sockets = append(sockets, tcpl)
+		}
 	}
 
 	d.Sockets = sockets
```

Comment:
simply set sockets to empty right here when running in mock mode and that'll save you all of the above indent ^